src/Microsoft.Azure.WebJobs.Host/Executors/FunctionExecutor.cs

```diff
@@ -499,11 +546,23 @@ private static ITaskSeriesTimer StartParameterLogTimer(IRecurrentCommand updateC
                 await singleton.AcquireAsync(functionCancellationTokenSource.Token);
             }
 
-            var timer = StartFunctionTimeout(instance, globalFunctionTimeout, functionCancellationTokenSource, traceWriter);
+            bool hasTimerElapsed = false;
+            var timer = StartFunctionTimeout(instance, timeoutConfig.Timeout, timeoutConfig.TimeoutWhileDebugging, functionCancellationTokenSource, traceWriter, () => hasTimerElapsed = true);
             try
             {
                 await invoker.InvokeAsync(invokeParameters);
             }
+            catch (OperationCanceledException)
+            {
+                // if cancellationToken is not signaled, it means this is a timeout
+                if (timeoutConfig.ThrowExceptionOnFunctionTimeout && hasTimerElapsed)
+                {
+                    string errorMessage = string.Format("Timeout while executing function: {0}", instance.FunctionDescriptor.ShortName);
```

Comment:
nit: Could this message include a) how long we ran; b) what the timeout threshold is?  In lots of cases, we only have an exception msg, and that could be critical info.